plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordSet.java

```diff
@@ -60,6 +71,6 @@ public List<Type> getColumnTypes()
     @Override
     public RecordCursor cursor()
     {
-        return new JdbcRecordCursor(jdbcClient, executor, session, split, table, columnHandles);
+        return retry(policy, () -> new JdbcRecordCursor(jdbcClient, executor, session, split, table, columnHandles));
```

Comment:
Question here, why we do not use `RetryJdbcClient` directly
The `JdbcRecordCursor` will use the `RetryJdbcClient` itself.

Reply:
`jdbcClient` is already here wrapped with `RetryingJdbcClient`. The issue here is that `JdbcRecordCursor` is using `jdbcClient` in a way that it is not enough. For example you cannot retry jdbc query using the same connection. The connection could be already invalid. Hence we need to retry on higher level.